tests/Model/OrderAdjustmentsRecalculationTest.php

```diff
@@ -0,0 +1,178 @@
+    {
+        $adjustment = $this->createAdjustment(AdjustmentInterface::ORDER_ITEM_PROMOTION_ADJUSTMENT, -300);
+        $this->item->addAdjustment($adjustment);
+        $this->item->removeAdjustment($adjustment);
+
+        $this->assertEquals(2000, $this->order->getTotal());
+        $this->assertEquals(0, $this->item->getAdjustmentsTotal());
+    }
+
+    /** @test */
+    public function it_recalculates_order_properly_while_order_promotion_adjustment_removed(): void
+    {
+        $adjustmentAmount = 100;
+        $adjustment = $this->createAdjustment(AdjustmentInterface::ORDER_PROMOTION_ADJUSTMENT, -100);
+        $this->order->addAdjustment($adjustment);
+        $this->order->removeAdjustment($adjustment);
+
+        $this->assertEquals(2000, $this->order->getTotal());
+        $this->assertEquals(0, $this->order->getAdjustmentsTotal());
+    }
+
+    /** @test */
+    public function it_recalculates_order_properly_while_order_shipping_promotion_adjustment_removed(): void
+    {
+        $adjustment = $this->createAdjustment(AdjustmentInterface::ORDER_SHIPPING_PROMOTION_ADJUSTMENT, -50);
+        $this->order->addAdjustment($adjustment);
+        $this->order->removeAdjustment($adjustment);
+
+        $this->assertEquals(2000, $this->order->getTotal());
+        $this->assertEquals(0, $this->order->getAdjustmentsTotal());
+    }
+
+    /** @test */
+    public function it_recalculates_order_properly_while_order_unit_promotion_adjustment_removed(): void
+    {
+        $adjustment = $this->createAdjustment(AdjustmentInterface::ORDER_UNIT_PROMOTION_ADJUSTMENT, -200);
+        $this->unitNumberOne->addAdjustment($adjustment);
+        $this->unitNumberTwo->addAdjustment($adjustment);
+        $this->unitNumberOne->removeAdjustment($adjustment);
+
+        $this->assertEquals(1800, $this->order->getTotal());
+        $this->assertEquals(0, $this->unitNumberOne->getAdjustmentsTotal());
+        $this->assertEquals(-200, $this->unitNumberTwo->getAdjustmentsTotal());
+    }
+
+    /** @test */
+    public function it_recalculates_order_properly_while_shipping_adjustment_removed(): void
+    {
+        $adjustment = $this->createAdjustment(AdjustmentInterface::SHIPPING_ADJUSTMENT, 75);
+        $this->order->addAdjustment($adjustment);
+        $this->order->removeAdjustment($adjustment);
+
+        $this->assertEquals(2000, $this->order->getTotal());
+        $this->assertEquals(0, $this->order->getAdjustmentsTotal());
+    }
+
+    /** @test */
+    public function it_recalculates_order_properly_while_tax_adjustment_removed(): void
+    {
+        $adjustment = $this->createAdjustment(AdjustmentInterface::TAX_ADJUSTMENT, 90);
+        $this->order->addAdjustment($adjustment);
+        $this->order->removeAdjustment($adjustment);
+
+        $this->assertEquals(2000, $this->order->getTotal());
+        $this->assertEquals(0, $this->order->getAdjustmentsTotal());
+    }
```

Comment:
I don't see a significant value coming from testing every type of adjustment. Let's create a simple test case where you have some adjustment already, you check if the returned value is correct, then you add 2–3 adjustments (both a negative and a positive) and again check if it's recalculated. We don't need anything more.

At the same time, you don't have such tests for `OrderItem` and `OrderItemUnit`.

Reply:
There are tests for `OrderItem` `OrderItemUnit` it is part of order
https://github.com/Sylius/Sylius/pull/12502/files#diff-e1c288738071c3d116f3735eead129fe90f38e7d8fd51e28f7bad8a738f03d0aR69-R78
https://github.com/Sylius/Sylius/pull/12502/files#diff-e1c288738071c3d116f3735eead129fe90f38e7d8fd51e28f7bad8a738f03d0aR39-R46

Reply:
I would be after modifying this tests by adding adjustments that already applies on order that im creating in SetUp, WDYT?